### ISSUE TITLE
Unable to boot with kernel-latest-4.15.6-1

### ISSUE THREAD

**Issue by coeusite** [QubesOS/qubes-issues#3790]
### Qubes OS version:

R4.0-rc5 with updates from current-testing repo

### Affected component(s): 

kernel

---

### Steps to reproduce the behavior:

The build:
- AMD R5 2400
- Gigabytes AB350N ITX

To reproduce this issue:
- install kernel-latest for dom0 from current-testing repo
- switch to kernel-latest-4.15.6-1
- boot system

### Expected behavior:

- boot up as usual

### Actual behavior:

- booting process freezes just before starting sys-net or any other Qubes
- no response for pressing keys 

### General notes:

- kernel-4.14.18-1 does not have this issue
- kernel-latest-4.15.6-1 does not have this issue on an Intel Z270 build

---

### Related issues:

The GPU of raven ridge APU requires kernel 4.15 or above to work properly, otherwise the only display resolution is 1600x1200. I, thus, wanna try kernel-latest package, but this bug stops me from doing so.

**Comment by JohnDoe02** [comment]
It had the same issue, when I tried to install Qubes on the HP Envy x360. I fixed the missing firmware files by installing the firmware package of the (still in development) Fedora 28.

However, with kernel 4.15, the Envy would almost always freeze during boot (with 4.14 it would boot, but Xorg wouldn't start). Compiling a kernel from drm-next fixed that, however, I had heavy graphical glitches which made the system unusable.

So it seems that the support for Raven Ridge is still somewhat buggy, good look for your efforts.

**Comment by marmarek** [comment]
There is updated kernel-latest package - 4.16.2. Does is still happen there?

**Comment by Aekez** [comment]
Hello, @marmarek 
I have the exact same issue as @JohnDoe02 and @coeusite on an HP Envy x360 (Ryzen 5, 2500U APU).

I can confirm it still happens with the Linux-Firmware package from Qubes unstable repository which includes the Raven driver for AMDGPU, and any of the available 4.15+ Qubes kernels.

**Update details**
I've tested all available Qubes 4.15+ kernels, and compiled the 4.16 kernel my self too, they all gave the same error results.
- linux-firmware-20180402-83.git8c1e439c.qubes1.fc25.noarch.rpm
- kernel-latest-4.15.1-1.pvops.qubes.x86_64.rpm
- kernel-latest-4.15.6-1.pvops.qubes.x86_64.rpm
- kernel-latest-4.16.2-1.pvops.qubes.x86_64.rpm
- kernel-latest-4.16.2-1.pvops.qubes.x86_64.rpm (Compiled with CONFIG_DRM_AMDGPU=y).

**Most of my tests are from an installed Qubes and not the installer**
To clarify, the installer also fail to install with graphics, but I either install Qubes with a kickstart file in text mode, or I transfer a fully functional fresh installed non-encrypted Qubes OS system over with the DD command from another working Qubes machine, following up with dracut and grub2-mkconfig.

**The new problem emerging after driver and kernel update**
Doing the above steps seems to solve the initial problems with the lack of driver and dependencies, xorg also no longer printed errors during boot. But now instead it lead to kernel panics, and the graphics never loaded. I never got as far as JohnDoe02 or coeusite, mine always resulted in kernel panic before any graphics can load, no matter which 4.15+ kernel or kernel settings I tried. The odd part is the kernel panics location during the boot process seems to change very easily by changing settings, such as acpi=off and amdgpu.dc=1, and similar related commands, but it's hard to find any patterns. It does kind of look like memory/power management issues in either Xen or the AMDGPU driver? But It's outside my expertise to figure out if that truly is the case. Looking through the logs for Xen, I found some errors toward and near the end of the logs, but I'm not sure how relevant they are. 

**/var/logs for HP Envy x360 (Ryzen 5, 2500U APU)** 
[hypervisor.log](https://github.com/QubesOS/qubes-issues/files/1993648/hypervisor.log)
[boot.log](https://github.com/QubesOS/qubes-issues/files/1993670/boot.log)
[Xorg.9.log](https://github.com/QubesOS/qubes-issues/files/1993671/Xorg.9.log)


**Other OS systems work fine on the laptop**
- Successfully installed Fedora-27 (Ran smooth while testing, a couple of days).
- Successfully installed Fedora-28 (Ran smooth while testing, a day or so).
- Original install Windows 10 for the brief time I had it installed ran pretty decently too, 
  - Also worked fine when re-installed to update to latest F17 Rev.A BIOS. 

I've been speculating about copying the Fedora-27 kernel config settings and try adjust kernel-latest-4.16.2-1.pvops.qubes.x86_64, but I haven't gotten around to it yet, it's on my list to try. I do unfortunately not remember where I found this minor clue, but the NVMe drive's driver module is also on my suspect list. I'm wondering if I should break warranty to insert a regular SSD instead, but I'm not quite there yet. Booting an existing installed Qubes from an USB drive failed miserably, seemingly because the NVMe somehow messes it up, but I have not confirmed that yet.

It's a lot of small clues and no real suspect, but for now I get the feeling that it's either Xen or AMDGPU bug related. But since it works on Fedora-27/28, it might rather be Xen causing issues? Could it be? I'll happily provide more logs if you need any. For now I transcribed one of the typical boot errors down below, when the kernel panic occurs. 


**The kernel panic seen on screen during boot**
- (( )) messages below are my own inserted notes,
- (~) represents memory/code address code-lines (bunch of numbers and letters), 
- I'll gladly edit and include modules and (~) values if they are needed to debug.

```
((Starting from the usual green (OK) messages during Qubes boot))
- Starting The Xen xenstore... (OK)
- Starting Qubes memory management... (OK)
- Starting Daemon for power management... (OK)
- Started D-Bus System Message Bus. (OK)
- general protection fault: 0000 [s1] SMP MOPTI
- Modules linked in: btusb btrtl btbcm btintel, and the list goes on for several lines of various different modules. 
((Oddly it posts a second output (two outputs with a list of modules) without any explanation as to why))
- CPU: 0 PID: 8 Comm: swapper/0 Tainted: G       WC      4.16.2.-1.pvops.qubes.x86_64 #1 
- Hardware name: HP HP ENVY x360 Convertible
- RIP: e838:switch_mm:irqs_off+8xe2/8x510
((A whole bunch of code addresses specified by letters and numbers that I don't believe is needed, referenced as (~), however I'll write them down if they can be useful to debug.))
- RSP: (e02b:ffffc900017c7940) EFLAGS: (~) ORIG_RAX: (~)
- RAX: (~) RBX: (~) RCX: (~)
- RDX: (~) RSI: (~) RDI: (~)
- RBP: (~) R08: (~) R09: (~)
- R10: (~) R11: (~) R12: (~)
- R13: (~) R14: (~) R15: (~)
- FS: (~) GS: (~) knIGS: (~)
- CS: (~) ES: (~) CR0: (~)
- CR2: (~) CR3: (~) CR4: (~)
- Call Trace:
__shcedule+0x39d/0x880
? hrtimer_start_range_ns+0x199/0x2c0
schedule+0x32/0x80
schedule_hrtimeout_range_clock+8xb9/0x1a0
? __hrtimer_init+0xb0/0xb0
poll_schdule_timeout+0x41/0x70
do_sys_poll+0x514/0x5d0
? _raw_spin_unlock_irqrestore+0x16/0x20
? ep_poll_callback+0x11d/0x2e0
? __alloc_skb+0x96/0x1c0
? __wake_up_common+0x96/0x100
? _raw_spin_unlock_irqrestore+0x16/0x20
? __wake_up_common_lock+0x09/0xc0
? cond_resched+0x16/0x40
? mutex_lock+0xe/0x30
? unix_stream_read_generic+0x225/0x0e0
? compact_poll_select_copy_remaining+0x140/0x140
? import_iovoc+0x37/0xd0
? unix_stream_recvmsg+0x53/0x230
? unix_state_double_unlock+0x40/0x40
? ___sys_recvmsg+0x18b/0x230
? seccomp_run_filters+0x58/0xb0
? xen_set_pte_at+0x8d/0x520
? __seccomp_filter+0x59/0x170
? ktime_get_ts64+0x58/0x520
? sys_ppoll+0x162/0x180
- SyS_ppoll+0x162/0x180
- do_syscall_64+0x74/0x180
- entry_SYSCALL_64_after_hwframe+0x3d/0xa2
- RIP: 00xx:0x776c0fa3dfad
((Below different numbers to the above numbers and letters, I'll edit to include if needed))
- RSP: (002b:00007ffe225f0f80) EFLAGS: (~) ORIG_RAX: (~)
- RAX: (~) RBX: (~) RCX: (~)
- RDX: (~) RSI: (~) RDI: (~)
- RBP: (~) R08: (~) R09: (~)
- R10: (~) R11: (~) R12: (~)
- R13: (~) R14: (~) R15: (~)
- Code: 85 c0 74 29 48 3b 90 e8 +2 ++ ++ 74 20 48 8b 80 70 03 00 00 83 e0 03 83 83f8 01 74 11b9 49 00 00 00 b8 01 00 00 00 ba 00 00 00 00 <0f> 30 48 89 e0 48 8b 53 50 48 c1 e8 27 25 ff 01 00 00 48 8d 3c
- RIP: switch_mm_irqs_off+0xe2/0x510 RSP: ffffc900017c7940
---[ end trace b98f9ba97a6ed38 ]---
- Kernel panic - not syncing: Fatal exception 
- Kernel Offset: disabled 
```

Screen stays like a freeze for a second or two, and then goes into a reboot. Repeat in an endless circle.

**Notes:**
1) In regard to what coeusite said about no responses from keys, I noticed this happening to me too when I use the `acpi=off` command. This is specifically on the 4.14 kernel, I never get enough time to verify this on kernel 4.15+ kernels due to the kernel panic.

**Comment by coeusite** [comment]
@Aekez Actually, the installer's graphics is not working for amd a8-9600 (Bristol Ridge), so probably it is not working for all AM4 PC. However, I can simply boot a flesh encrypted-Qubes installation with an AMD AM4 PC (QubesOS on USB SSD).

Currently I use an Intel PC for my Qubes system.

**Comment by Aekez** [comment]
Turns out I had given up too early on the external USB approach, I went back and tried your external drive suggestion and to my surprise managed to get it to work! It's very, very odd though, the other approaches should have worked if this works, so I'm a bit puzzled. However it's not the first time I had issues with installing Qubes on systems with a NVMe drive, where I had to move the NVMe to another machine, install Qubes, and then move it back to the original machine. Perhaps this is the same? I'm surprised if it is though since it acted very different from last time (the kernel panics and all). 

In case any readers find this issue and need to know the steps to get it from USB drive to the laptops SSD/NVMe/HDD drive, see steps below. I realize it may not be the perfect approach, but it works nonetheless. 

Check-list 
- Install Qubes on another drive on a computer Qubes works on.

**Comment by 0spinboson** [comment]
Strange. I recently had to reinstall Qubes R4.0 on my self-built x370 prime pro, r5 1600, rx vega 56, Intel 660p nvme system, and aside from the video card not being supported by the included 4.14 kernel, I had no issues at all. But yeah, RR support in 4.15 (and maybe also 4.16?) was a known issue, from what I remember reading on /r/amd.

**Comment by marmarek** [comment]
Does it still apply to recent kernel-latest? The one in stable repo is at 4.20.x, in testing at 5.0.x.

**Comment by 0spinboson** [comment]
I think this one can be closed @andrewdavidwong

**Comment by jamke** [comment]
I have this issue with AMD eGPU on R4.1 with `kernel-latest` from the default repo. The system is not bootable after the command `sudo qubes-dom0-update kernel-latest`.

I have Kernel panic after selecting default boot option in grub, the screen has a lot of text, like 2-3 screens and at the end:
```
Kernel panic - not syncing: Fatal exception
Kernel Offset: disabled
```

I can boot using iGPU when I remove eGPU and install anything or change anything.

How can that be fixed with `kernel-latest`?

**Comment by andrewdavidwong** [comment]
> I have this issue with AMD eGPU on R4.1 with `kernel-latest` from the default repo. [...]

I'm not sure if this issue should be reopened, as presumably your `kernel-latest` now refers to a different version than the versions discussed back when this issue was active (though you didn't mention your version, so I don't know that for certain).

My hunch is that it might be better to open a separate issue for this (which it seems you already did with #7380, right?).

@DemiMarie, what do you think?

**Comment by DemiMarie** [comment]
@andrewdavidwong @jamke agreed, this is a separate issue.

**Comment by jamke** [comment]
I agree, I should make a new issue with updated details.
Do not reopen that one, I was just hoping for fast reply with existing solution/work around if one exists because it looks like the same issue as mine.
